oxd-server/src/main/java/org/gluu/oxd/server/persistence/service/GluuPersistenceService.java

```diff
@@ -0,0 +1,306 @@
+    }
+
+    public void addBranch(String dn, String ouName) {
+        SimpleBranch branch = new SimpleBranch();
+        branch.setOrganizationalUnitName(ouName);
+        branch.setDn(dn);
+
+        this.persistenceEntryManager.persist(branch);
+    }
+
+    public boolean create(Rp rp) {
+        try {
+            RpObject rpObj = new RpObject(getDnForRp(rp.getOxdId()), rp.getOxdId(), Jackson2.serializeWithoutNulls(rp));
+            this.persistenceEntryManager.persist(rpObj);
+            LOG.debug("RP created successfully. RP : {} ", rp);
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to create RP: {} ", rp, e);
+        }
+        return false;
+    }
+
+    public boolean createExpiredObject(ExpiredObject obj) {
+        try {
+            if (isExpiredObjectPresent(obj.getKey())) {
+                LOG.warn("Expired_object already present. Object : {} ", obj.getKey());
+                return true;
+            }
+            obj.setTypeString(obj.getType().getValue());
+            obj.setDn(getDnForExpiredObj(obj.getKey()));
+            this.persistenceEntryManager.persist(obj);
+            LOG.debug("Expired_object created successfully. Object : {} ", obj.getKey());
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to create ExpiredObject: {} ", obj.getKey(), e);
+        }
+        return false;
+    }
+
+    public boolean update(Rp rp) {
+        try {
+            RpObject rpObj = new RpObject(getDnForRp(rp.getOxdId()), rp.getOxdId(), Jackson2.serializeWithoutNulls(rp));
+            this.persistenceEntryManager.merge(rpObj);
+            LOG.debug("RP updated successfully. RP : {} ", rpObj);
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to update RP: {} ", rp, e);
+        }
+        return false;
+    }
+
+    public Rp getRp(String oxdId) {
+        try {
+            RpObject rpFromGluuPersistance = getRpObject(oxdId, new String[0]);
+
+            Rp rp = MigrationService.parseRp(rpFromGluuPersistance.getData());
+            if (rp != null) {
+                LOG.debug("Found RP id: {}, RP : {} ", oxdId, rp);
+                return rp;
+            }
+            LOG.error("Failed to fetch RP by id: {} ", oxdId);
+            return null;
+        } catch (Exception e) {
+            LOG.error("Failed to update oxdId: {} ", oxdId, e);
+        }
+        return null;
+    }
+
+    private RpObject getRpObject(String oxdId, String... returnAttributes) {
+        return (RpObject) this.persistenceEntryManager.find(getDnForRp(oxdId), RpObject.class, returnAttributes);
+    }
+
+    public ExpiredObject getExpiredObject(String key) {
+        try {
+            ExpiredObject expiredObject = getExpiredObject(key, null);
+            expiredObject.setType(ExpiredObjectType.fromValue(expiredObject.getTypeString()));
+            if (expiredObject != null) {
```

Comment:
Here we check expiredObject on null but in line 185 we set type, which means we will get NPE if it's null. Should 185 line be moved down?

Reply:
Done